packages/auth/src/authClients/supabase.ts

```diff
@@ -32,13 +46,28 @@ export const supabase = (client: Supabase): AuthClientSupabase => {
   return {
     type: 'supabase',
     client,
-    login: ({ email, password }) => client.auth.signIn({ email, password }),
-    logout: () => client.auth.signOut(),
-    signup: ({ email, password }) => client.auth.signUp({ email, password }),
+    login: async ({ email, password, provider }) => {
+      if (email && !password) {
+        // magic link
+        return await client.auth.signIn({ email })
+      }
```

Comment:
Could we not make this a provider too? Then it's explicit rather than implicit.

So we add a value: "magiclink" and then branch on that.

Reply:
Supabase doesn't consider the magiclink "path" a provider -- it is email but passwordless.

If we were to add magiclink, then we would add "email" or "password" as well ... but this is not one of what the and GoTrue consider to be providers (aka oauth providers) listed:

https://github.com/supabase/gotrue-js/blob/7580c34aeb68d5e5b8d3548e0615801b7a9472d1/src/lib/types.ts#L1

In the signin docs: https://supabase.io/docs/client/auth-signin

```
* A user can sign up either via email or OAuth.
* If you provide email without a password, the user will be sent a magic link.
```

Since their rule was 

`If you provide email without a password, the user will be sent a magic link.`

that's how I chose to implement it.

If it was a provider, I would have to set the provider to null -- wlll I imagine I ignore it.

But regardless because 'magiclink' isn't in the GoTrue types

https://github.com/supabase/gotrue-js/blob/7580c34aeb68d5e5b8d3548e0615801b7a9472d1/src/lib/types.ts#L1

then 

`  * @param {'bitbucket' | 'github' | 'gitlab' | 'google' | 'azure'} options.provider One of the providers supported by GoTrue.`

wouldn't be valid.